### PR TITLE
Convert `echo` function to 4.x API

### PR DESCRIPTION
**Pull request by alexjfisher**
4.x functions don't have environment isolation issues.

### CODE REVIEW

File: lib/puppet/functions/echo.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# This function outputs the variable content and its type to the
+# debug log. It's similiar to the `notice` function but provides
+# a better output format useful to trace variable types and values
+# in the manifests.
+#
+# ```
```
Comment (alexjfisher) [inline lib/puppet/functions/echo.rb]: `@example` isn't implemented yet in puppet-strings :(

**Comment by ekohl** [comment]
By removing the old function, is this a breaking change?

**Comment by alexjfisher** [comment]
Puppet 3 support was dropped in 2.0.0, so I don't think this is breaking.  It would have been if I'd namespaced the function.  Perhaps we will namespace all functions (in a single breaking change) at a later date.

**Comment by ekohl** [comment]
@alexjfisher if someone uses the unnamespaced function, wouldn't that break their catalog compilation?

**Comment by ekohl** [comment]
I assumed the new API always namespaces but I was wrong.

**Pull request merged**